### DISCUSSION TITLE
Integrating TinyOBJLoader into MapLibre-Native

### DISCUSSION THREAD

**Discussion by hmtriit** [maplibre/maplibre-native#2295] [Q&A]
Hi Master,

I am integrating [TinyOBJLoader](https://github.com/tinyobjloader/tinyobjloader) into MapLibre-Native, but I cannot display the 3D model on in MapLibre-Native during the rendering step. Can you please guide me on how to achieve this?

Hope your help!

Thanks

**Comment by hmtriit** [comment]
Hi @louwers 
I am currently working on a project to display 3D models on a map interface, but I am encountering some difficulties in the rendering and geographical positioning stages. Although I have successfully loaded the data and can access it (3D building OBJ model + texture), I am unsure how to proceed with the next steps. 
Below are the steps I have implemented so far:
1. Cloned and included "tinyobjloader" in the "/mbgl/src/tinyobjloader" directory and built it using a build system like CMake  (Android) and Bazel (IOS).
2. Cloned the custom layer class for use with objModelLayer.
3. Defined the .hpp and .cpp files for the same custom layer.
4. Implemented tinyobjloader in the render() function.

I would greatly appreciate your assistance with the following tasks:
1. Rendering and displaying the 3D model on the map.
2. Converting and returning the model to latitude and longitude coordinates for accurate placement on the map. Could you please provide detailed guidance or refer me to relevant documentation that could assist in resolving these issues?

Thank you in advance for your help!

**Reply by louwers** [reply]
> tinyobjloader is good for embedding .obj loader to your (global illumination) renderer ;-)

I am not a graphics engineer, but MapLibre Native is not a global illumination renderer.

**Reply by hmtriit** [reply]
hi @louwers 
Is there someone on your team who can help with this?

Thanks

**Reply by louwers** [reply]
Wet Dog Weather is doing contracting work to work on MapLibre Native. You could reach out to them.

**Reply by WuMianzhi** [reply]
do you success? Could you please tell me how to got it?

**Comment by thothai94** [comment]
Look forward to MapLibre Native support 3D model layer. It will be a nice feature!

**Reply by louwers** [reply]
I know it's possible with MapLibre GL JS https://maplibre.org/maplibre-gl-js/docs/examples/add-3d-model/

I am not sure if anyone ever did it with MapLibre Native.
